posts/2020-10-23-beta-20.0.0.12.adoc

```diff
@@ -0,0 +1,452 @@
++
+The methods have been added to allow users to retrieve multi-valued properties as lists instead of arrays. The methods return the resolved property values with the specified `propertyType` for the specified `propertyName`.
+
+* `OptinalInt`, `OptionalLong` and `OptionalDouble` are now provided as built-in link:https://download.eclipse.org/microprofile/microprofile-config-2.0-RC1/microprofile-config-spec.html#converter[Converters]. The new Converters can be used like any of the other built-in Converters, converting injected Config property values to a defined type:
++
+[source, java]
+----
+@Inject
+@ConfigProperty(name = "my.optional.int.property")
+private OptionalInt optionalIntProperty;
+----
+
+=== Incompatible changes
+
+If you move up from MicroProfile Config 1.x to 2.0, please take care of the following incompatible changes:
+
+* `ConfigSource.getPropertyNames` is no longer a default method. The implementation of a `ConfigSource` must implement this method.
+* Previous versions of the specification would not evaluate Property Expressions. As such, previous working configuration may behave differently (if the used configuration contains values with Property Expressions syntax). Property Expressions can be disabled by setting the property `mp.config.property.expressions.enabled` with the value of `false`.
+* Empty values or other special characters are no longer valid Config property values. This means that `Config.getValue(String propertyName, Class<T> propertyType)` will never return `null`; a `NoSuchElementException` will be thrown if either a Config property value is not defined, is defined as an empty string, or its converter returns `null`. In the previous release, the empty value was returned as an empty value. For more information, please refer to this link:https://download.eclipse.org/microprofile/microprofile-config-2.0-RC1/microprofile-config-spec.html#_config_value_conversion_rules[section].
+
+To enable the new beta features in your app, add them to your `server.xml`:
+
+[source, xml]
+----
+<featureManager>
+   <feature>mpConfig-2.0</feature>
+</featureManager>
+----
+
+For more information:
+
+link:https://download.eclipse.org/microprofile/microprofile-config-2.0-RC1/microprofile-config-spec.html#release_notes_20[Release Notes for MicroProfile Config 2.0]
+
+link:https://github.com/eclipse/microprofile-config/milestone/8?closed=1[MicroProfile Config 2.0 Milestone]
+
+[#jwt]
+=== Support for MicroProfile JWT 1.2
+
+The MicroProfile JSON Web Token 1.2 specification allows using a JWT token for authenticating and authorizing requests to a service. The specification simplifies the configuration for managing the validation of the JWT by introducing new MicroProfile Config properties. Enhanced signature algorithm support is added in this Open Liberty beta.
+
+Version 1.2 of the MicroProfile JSON Web Token specification adds the following MicroProfile Config properties to control different aspects of the JWT validation.
```

Comment:
Change to "MicroProfile JSON Web Token 1.2 adds the following MicroProfile Config properties to control different aspects of the JWT validation."

(basically we rarely need to talk about "specification". It's just what the feature does. The fact that it's been implemented from a specification is an internal detail for us to care about, not for developers who use it to care about.)